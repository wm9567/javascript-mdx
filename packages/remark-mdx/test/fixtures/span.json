{
  "type": "root",
  "children": [
    {
      "type": "paragraph",
      "children": [
        {
          "type": "text",
          "value": "Hereâ€™s a sum: ",
          "position": {
            "start": {
              "line": 1,
              "column": 1,
              "offset": 0
            },
            "end": {
              "line": 1,
              "column": 15,
              "offset": 14
            }
          }
        },
        {
          "type": "inlineCode",
          "value": "1 + 1",
          "position": {
            "start": {
              "line": 1,
              "column": 15,
              "offset": 14
            },
            "end": {
              "line": 1,
              "column": 22,
              "offset": 21
            }
          }
        },
        {
          "type": "text",
          "value": " equals ",
          "position": {
            "start": {
              "line": 1,
              "column": 22,
              "offset": 21
            },
            "end": {
              "line": 1,
              "column": 30,
              "offset": 29
            }
          }
        },
        {
          "type": "mdxJsxTextElement",
          "name": null,
          "attributes": [],
          "children": [
            {
              "type": "mdxTextExpression",
              "value": "1 + 1",
              "position": {
                "start": {
                  "line": 1,
                  "column": 32,
                  "offset": 31
                },
                "end": {
                  "line": 1,
                  "column": 39,
                  "offset": 38
                }
              },
              "data": {
                "estree": {
                  "type": "BinaryExpression",
                  "start": 0,
                  "end": 5,
                  "loc": {
                    "start": {
                      "line": 1,
                      "column": 0
                    },
                    "end": {
                      "line": 1,
                      "column": 5
                    }
                  },
                  "left": {
                    "type": "Literal",
                    "start": 0,
                    "end": 1,
                    "loc": {
                      "start": {
                        "line": 1,
                        "column": 0
                      },
                      "end": {
                        "line": 1,
                        "column": 1
                      }
                    },
                    "value": 1,
                    "raw": "1"
                  },
                  "operator": "+",
                  "right": {
                    "type": "Literal",
                    "start": 4,
                    "end": 5,
                    "loc": {
                      "start": {
                        "line": 1,
                        "column": 4
                      },
                      "end": {
                        "line": 1,
                        "column": 5
                      }
                    },
                    "value": 1,
                    "raw": "1"
                  }
                }
              }
            }
          ],
          "position": {
            "start": {
              "line": 1,
              "column": 30,
              "offset": 29
            },
            "end": {
              "line": 1,
              "column": 42,
              "offset": 41
            }
          }
        },
        {
          "type": "text",
          "value": ".",
          "position": {
            "start": {
              "line": 1,
              "column": 42,
              "offset": 41
            },
            "end": {
              "line": 1,
              "column": 43,
              "offset": 42
            }
          }
        }
      ],
      "position": {
        "start": {
          "line": 1,
          "column": 1,
          "offset": 0
        },
        "end": {
          "line": 1,
          "column": 43,
          "offset": 42
        }
      }
    },
    {
      "type": "paragraph",
      "children": [
        {
          "type": "text",
          "value": "Welcome: ",
          "position": {
            "start": {
              "line": 3,
              "column": 1,
              "offset": 44
            },
            "end": {
              "line": 3,
              "column": 10,
              "offset": 53
            }
          }
        },
        {
          "type": "mdxJsxTextElement",
          "name": "div",
          "attributes": [],
          "children": [
            {
              "type": "text",
              "value": "Hello, world",
              "position": {
                "start": {
                  "line": 3,
                  "column": 15,
                  "offset": 58
                },
                "end": {
                  "line": 3,
                  "column": 27,
                  "offset": 70
                }
              }
            }
          ],
          "position": {
            "start": {
              "line": 3,
              "column": 10,
              "offset": 53
            },
            "end": {
              "line": 3,
              "column": 33,
              "offset": 76
            }
          }
        },
        {
          "type": "text",
          "value": "!",
          "position": {
            "start": {
              "line": 3,
              "column": 33,
              "offset": 76
            },
            "end": {
              "line": 3,
              "column": 34,
              "offset": 77
            }
          }
        }
      ],
      "position": {
        "start": {
          "line": 3,
          "column": 1,
          "offset": 44
        },
        "end": {
          "line": 3,
          "column": 34,
          "offset": 77
        }
      }
    },
    {
      "type": "paragraph",
      "children": [
        {
          "type": "text",
          "value": "OR as an expression: ",
          "position": {
            "start": {
              "line": 5,
              "column": 1,
              "offset": 79
            },
            "end": {
              "line": 5,
              "column": 22,
              "offset": 100
            }
          }
        },
        {
          "type": "mdxJsxTextElement",
          "name": "div",
          "attributes": [],
          "children": [
            {
              "type": "text",
              "value": "Hello, ",
              "position": {
                "start": {
                  "line": 5,
                  "column": 27,
                  "offset": 105
                },
                "end": {
                  "line": 5,
                  "column": 34,
                  "offset": 112
                }
              }
            },
            {
              "type": "mdxTextExpression",
              "value": "formatName(user)",
              "position": {
                "start": {
                  "line": 5,
                  "column": 34,
                  "offset": 112
                },
                "end": {
                  "line": 5,
                  "column": 52,
                  "offset": 130
                }
              },
              "data": {
                "estree": {
                  "type": "CallExpression",
                  "start": 0,
                  "end": 16,
                  "loc": {
                    "start": {
                      "line": 1,
                      "column": 0
                    },
                    "end": {
                      "line": 1,
                      "column": 16
                    }
                  },
                  "callee": {
                    "type": "Identifier",
                    "start": 0,
                    "end": 10,
                    "loc": {
                      "start": {
                        "line": 1,
                        "column": 0
                      },
                      "end": {
                        "line": 1,
                        "column": 10
                      }
                    },
                    "name": "formatName"
                  },
                  "arguments": [
                    {
                      "type": "Identifier",
                      "start": 11,
                      "end": 15,
                      "loc": {
                        "start": {
                          "line": 1,
                          "column": 11
                        },
                        "end": {
                          "line": 1,
                          "column": 15
                        }
                      },
                      "name": "user"
                    }
                  ],
                  "optional": false
                }
              }
            },
            {
              "type": "text",
              "value": "!",
              "position": {
                "start": {
                  "line": 5,
                  "column": 52,
                  "offset": 130
                },
                "end": {
                  "line": 5,
                  "column": 53,
                  "offset": 131
                }
              }
            }
          ],
          "position": {
            "start": {
              "line": 5,
              "column": 22,
              "offset": 100
            },
            "end": {
              "line": 5,
              "column": 59,
              "offset": 137
            }
          }
        },
        {
          "type": "text",
          "value": ".",
          "position": {
            "start": {
              "line": 5,
              "column": 59,
              "offset": 137
            },
            "end": {
              "line": 5,
              "column": 60,
              "offset": 138
            }
          }
        }
      ],
      "position": {
        "start": {
          "line": 5,
          "column": 1,
          "offset": 79
        },
        "end": {
          "line": 5,
          "column": 60,
          "offset": 138
        }
      }
    },
    {
      "type": "paragraph",
      "children": [
        {
          "type": "text",
          "value": "Attributes: ",
          "position": {
            "start": {
              "line": 7,
              "column": 1,
              "offset": 140
            },
            "end": {
              "line": 7,
              "column": 13,
              "offset": 152
            }
          }
        },
        {
          "type": "mdxJsxTextElement",
          "name": "div",
          "attributes": [
            {
              "type": "mdxJsxAttribute",
              "name": "tabIndex",
              "value": "0"
            }
          ],
          "children": [],
          "position": {
            "start": {
              "line": 7,
              "column": 13,
              "offset": 152
            },
            "end": {
              "line": 7,
              "column": 37,
              "offset": 176
            }
          }
        }
      ],
      "position": {
        "start": {
          "line": 7,
          "column": 1,
          "offset": 140
        },
        "end": {
          "line": 7,
          "column": 37,
          "offset": 176
        }
      }
    },
    {
      "type": "paragraph",
      "children": [
        {
          "type": "text",
          "value": "Attribute value expressions: ",
          "position": {
            "start": {
              "line": 9,
              "column": 1,
              "offset": 178
            },
            "end": {
              "line": 9,
              "column": 30,
              "offset": 207
            }
          }
        },
        {
          "type": "mdxJsxTextElement",
          "name": "img",
          "attributes": [
            {
              "type": "mdxJsxAttribute",
              "name": "src",
              "value": {
                "type": "mdxJsxAttributeValueExpression",
                "value": "user.avatarUrl",
                "data": {
                  "estree": {
                    "type": "MemberExpression",
                    "start": 0,
                    "end": 14,
                    "loc": {
                      "start": {
                        "line": 1,
                        "column": 0
                      },
                      "end": {
                        "line": 1,
                        "column": 14
                      }
                    },
                    "object": {
                      "type": "Identifier",
                      "start": 0,
                      "end": 4,
                      "loc": {
                        "start": {
                          "line": 1,
                          "column": 0
                        },
                        "end": {
                          "line": 1,
                          "column": 4
                        }
                      },
                      "name": "user"
                    },
                    "property": {
                      "type": "Identifier",
                      "start": 5,
                      "end": 14,
                      "loc": {
                        "start": {
                          "line": 1,
                          "column": 5
                        },
                        "end": {
                          "line": 1,
                          "column": 14
                        }
                      },
                      "name": "avatarUrl"
                    },
                    "computed": false,
                    "optional": false
                  }
                }
              }
            }
          ],
          "children": [],
          "position": {
            "start": {
              "line": 9,
              "column": 30,
              "offset": 207
            },
            "end": {
              "line": 9,
              "column": 62,
              "offset": 239
            }
          }
        },
        {
          "type": "text",
          "value": ".",
          "position": {
            "start": {
              "line": 9,
              "column": 62,
              "offset": 239
            },
            "end": {
              "line": 9,
              "column": 63,
              "offset": 240
            }
          }
        }
      ],
      "position": {
        "start": {
          "line": 9,
          "column": 1,
          "offset": 178
        },
        "end": {
          "line": 9,
          "column": 63,
          "offset": 240
        }
      }
    },
    {
      "type": "paragraph",
      "children": [
        {
          "type": "text",
          "value": "Attribute expressions: ",
          "position": {
            "start": {
              "line": 11,
              "column": 1,
              "offset": 242
            },
            "end": {
              "line": 11,
              "column": 24,
              "offset": 265
            }
          }
        },
        {
          "type": "mdxJsxTextElement",
          "name": "div",
          "attributes": [
            {
              "type": "mdxJsxExpressionAttribute",
              "value": "...props",
              "data": {
                "estree": {
                  "type": "SpreadElement",
                  "start": 3,
                  "end": 11,
                  "loc": {
                    "start": {
                      "line": 1,
                      "column": 3
                    },
                    "end": {
                      "line": 1,
                      "column": 11
                    }
                  },
                  "argument": {
                    "type": "Identifier",
                    "start": 6,
                    "end": 11,
                    "loc": {
                      "start": {
                        "line": 1,
                        "column": 6
                      },
                      "end": {
                        "line": 1,
                        "column": 11
                      }
                    },
                    "name": "props"
                  }
                }
              }
            }
          ],
          "children": [],
          "position": {
            "start": {
              "line": 11,
              "column": 24,
              "offset": 265
            },
            "end": {
              "line": 11,
              "column": 46,
              "offset": 287
            }
          }
        }
      ],
      "position": {
        "start": {
          "line": 11,
          "column": 1,
          "offset": 242
        },
        "end": {
          "line": 11,
          "column": 46,
          "offset": 287
        }
      }
    },
    {
      "type": "paragraph",
      "children": [
        {
          "type": "text",
          "value": "Self-closing elements: ",
          "position": {
            "start": {
              "line": 13,
              "column": 1,
              "offset": 289
            },
            "end": {
              "line": 13,
              "column": 24,
              "offset": 312
            }
          }
        },
        {
          "type": "mdxJsxTextElement",
          "name": "img",
          "attributes": [],
          "children": [],
          "position": {
            "start": {
              "line": 13,
              "column": 24,
              "offset": 312
            },
            "end": {
              "line": 13,
              "column": 30,
              "offset": 318
            }
          }
        }
      ],
      "position": {
        "start": {
          "line": 13,
          "column": 1,
          "offset": 289
        },
        "end": {
          "line": 13,
          "column": 30,
          "offset": 318
        }
      }
    },
    {
      "type": "paragraph",
      "children": [
        {
          "type": "text",
          "value": "Multiple elements: ",
          "position": {
            "start": {
              "line": 15,
              "column": 1,
              "offset": 320
            },
            "end": {
              "line": 15,
              "column": 20,
              "offset": 339
            }
          }
        },
        {
          "type": "mdxJsxTextElement",
          "name": "div",
          "attributes": [],
          "children": [
            {
              "type": "text",
              "value": "\n",
              "position": {
                "start": {
                  "line": 15,
                  "column": 25,
                  "offset": 344
                },
                "end": {
                  "line": 16,
                  "column": 1,
                  "offset": 345
                }
              }
            },
            {
              "type": "mdxJsxTextElement",
              "name": "h1",
              "attributes": [],
              "children": [
                {
                  "type": "text",
                  "value": "Hello!",
                  "position": {
                    "start": {
                      "line": 16,
                      "column": 7,
                      "offset": 351
                    },
                    "end": {
                      "line": 16,
                      "column": 13,
                      "offset": 357
                    }
                  }
                }
              ],
              "position": {
                "start": {
                  "line": 16,
                  "column": 3,
                  "offset": 347
                },
                "end": {
                  "line": 16,
                  "column": 18,
                  "offset": 362
                }
              }
            },
            {
              "type": "text",
              "value": "\n",
              "position": {
                "start": {
                  "line": 16,
                  "column": 18,
                  "offset": 362
                },
                "end": {
                  "line": 17,
                  "column": 1,
                  "offset": 363
                }
              }
            },
            {
              "type": "mdxJsxTextElement",
              "name": "h2",
              "attributes": [],
              "children": [
                {
                  "type": "text",
                  "value": "Good to see you here.",
                  "position": {
                    "start": {
                      "line": 17,
                      "column": 7,
                      "offset": 369
                    },
                    "end": {
                      "line": 17,
                      "column": 28,
                      "offset": 390
                    }
                  }
                }
              ],
              "position": {
                "start": {
                  "line": 17,
                  "column": 3,
                  "offset": 365
                },
                "end": {
                  "line": 17,
                  "column": 33,
                  "offset": 395
                }
              }
            },
            {
              "type": "text",
              "value": "\n",
              "position": {
                "start": {
                  "line": 17,
                  "column": 33,
                  "offset": 395
                },
                "end": {
                  "line": 18,
                  "column": 1,
                  "offset": 396
                }
              }
            }
          ],
          "position": {
            "start": {
              "line": 15,
              "column": 20,
              "offset": 339
            },
            "end": {
              "line": 18,
              "column": 7,
              "offset": 402
            }
          }
        },
        {
          "type": "text",
          "value": ".",
          "position": {
            "start": {
              "line": 18,
              "column": 7,
              "offset": 402
            },
            "end": {
              "line": 18,
              "column": 8,
              "offset": 403
            }
          }
        }
      ],
      "position": {
        "start": {
          "line": 15,
          "column": 1,
          "offset": 320
        },
        "end": {
          "line": 18,
          "column": 8,
          "offset": 403
        }
      }
    }
  ],
  "position": {
    "start": {
      "line": 1,
      "column": 1,
      "offset": 0
    },
    "end": {
      "line": 19,
      "column": 1,
      "offset": 404
    }
  }
}
