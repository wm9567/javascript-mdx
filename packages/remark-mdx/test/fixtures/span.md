Hereâ€™s a sum: `1 + 1` equals <>{1 + 1}</>.

Welcome: <div>Hello, world</div>!

OR as an expression: <div>Hello, {formatName(user)}!</div>.

Attributes: <div tabIndex="0"></div>

Attribute value expressions: <img src={user.avatarUrl}></img>.

Attribute expressions: <div {...props}></div>

Self-closing elements: <img/>

Multiple elements: <div>
  <h1>Hello!</h1>
  <h2>Good to see you here.</h2>
</div>.
