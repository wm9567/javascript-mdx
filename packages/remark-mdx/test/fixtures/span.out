Hereâ€™s a sum: `1 + 1` equals <>{1 + 1}</>.

Welcome: <div>Hello, world</div>!

OR as an expression: <div>Hello, {formatName(user)}!</div>.

Attributes: <div tabIndex="0"/>

Attribute value expressions: <img src={user.avatarUrl}/>.

Attribute expressions: <div {...props}/>

Self-closing elements: <img/>

Multiple elements: <div>
<h1>Hello!</h1>
<h2>Good to see you here.</h2>
</div>.
